### PR TITLE
Fixes to recording

### PR DESCRIPTION
**Pull request by Arri98**
**Description**

Fixed some issues with recording:

- Licode would not record streams that didn't have both video and audio. This was fixed by @[zafergurel](https://github.com/zafergurel) in #1265. Made minor changes for the linter so all credits to him.
- When stopping the recording, a wrong ID was passed resulting in the stream continued being recorded. This was fixed by @[nvazquezg](https://github.com/nvazquezg) in #1565
-  When checking if the stream had audio the condition if (getAudioSinkSSRC() == 0) was always returning false even if the stream had audio. In this if block the codec was set to PCMU so if other codec was used the resulting audio is corrupted. 

Also removed the toggleRecording button in the basic example and changed it for individual buttons so the user can select the stream to record.

[] It needs and includes Unit Tests

**Changes in Client or Server public APIs**

No changes

[] It includes documentation for these changes in `/doc`.
No changes

### CODE REVIEW

File: erizo_controller/erizoJS/models/Publisher.js [hunk 1]
```diff
@@ -98,11 +98,16 @@ class Source extends NodeClass {
...
     const eoId = `${url}_${this.streamId}`;
+    const hasVideo = options.hasVideo === null || options.hasVideo;
+    const hasAudio = options.hasAudio === null || options.hasAudio;
+
     log.info(`message: Adding ExternalOutput, id: ${eoId}, url: ${url},`,
       logger.objectToLog(this.options), logger.objectToLog(this.options.metadata));
     const externalOutput = new erizo.ExternalOutput(this.threadPool, url,
       Helpers.getMediaConfiguration(options.mediaConfiguration));
     externalOutput.id = eoId;
+    externalOutput.setHasAudioAndVideo(hasAudio, hasVideo);
```
Comment (lodoyun) [inline erizo_controller/erizoJS/models/Publisher.js:109]: Why not just pass `hasAudio` and `hasVideo` in the constructor just like we do for MediaStream?

**Review by lodoyun — COMMENTED** [review]
In general I think this looks good. I do have a question that I feel would avoid adding new methods to set the hasVideo and hasAudio flags when we already have that info when the object is created

File: erizo_controller/erizoJS/models/Publisher.js (hunk 1 above)
Comment (Arri98) [reply erizo_controller/erizoJS/models/Publisher.js]: Makes sense, I will update it

File: erizo_controller/erizoAgent/log4cxx.properties [hunk 2]
```diff
@@ -34,7 +34,7 @@ log4j.logger.dtls.DtlsSocketContext=WARN
 log4j.logger.dtls.SSL=WARN
 
 log4j.logger.media.ExternalInput=WARN
-log4j.logger.media.ExternalOutput=WARN
+log4j.logger.media.ExternalOutput=DEBUG
```
Comment (lodoyun) [inline erizo_controller/erizoAgent/log4cxx.properties:37]: consider making this `WARN` again

**Review by lodoyun — APPROVED** [review]
Looks good to me 👍  but please fix the log level as indicated in the comment
Thanks!

**Comment by zafergurel** [comment]
Thanks @Arri98 for giving credit and for your contribution.

**Pull request merged**